qt/scientific_interfaces/Direct/ALFView_view.cpp

```diff
@@ -0,0 +1,69 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ALFView_view.h"
+
+#include <QFileDialog>
+#include <QGridLayout>
+#include <QLineEdit>
+#include <QRegExpValidator>
+#include <QSplitter>
+#include <QVBoxLayout>
+
+namespace MantidQt {
+namespace CustomInterfaces {
+
+ALFView_view::ALFView_view(QWidget *parent)
+    : QWidget(parent), m_run(nullptr), m_loadRunObservable(nullptr),
+      m_browseObservable(nullptr) {
+  QSplitter *MainLayout = new QSplitter(Qt::Vertical, this);
+  QWidget *loadBar = new QWidget();
```

Comment:
We have the `MWRunFiles` widget in common widgets for doing file finding. Can we use this instead of rolling our own? It does all of the browsing, threading the file finding etc for you.

Reply:
We only ever want a single file to be loaded and I believe that `MWRunFiles`  allows multiple files

Reply:
I have had a look and you can set it to do single files and I can set the instrument. What I have not seen is a way to set the output ws.

Reply:
So the widget doesn't actually do the loading it just deals with finding the files, in a background thread, and providing the browse capability without everyone needing to reimplement it. The loading has to be handled separately.

Reply:
Ok, that sounds good. Is there a good example I could look at? I am trying to find one and having some difficulty.

Reply:
I have found one in muon analysis :)